draft-ietf-quic-transport.md

```diff
@@ -2228,28 +2230,28 @@ Note:
   control, which are not expected to be relevant for a closed connection.
   Retransmitting the final packet requires less state.
 
-After receiving a closing frame, endpoints enter the draining state.  An
-endpoint that receives a closing frame MAY send a single packet containing a
-closing frame before entering the draining state, using a CONNECTION_CLOSE frame
-and a NO_ERROR code if appropriate.  An endpoint MUST NOT send further packets,
-which could result in a constant exchange of closing frames until the closing
-period on either peer ended.
+After receiving a CONNECTION_CLOSE frame, endpoints enter the draining state.
+An endpoint that receives a CONNECTION_CLOSE frame MAY send a single packet
+containing a CONNECTION_CLOSE frame before entering the draining state, using a
+CONNECTION_CLOSE frame and a NO_ERROR code if appropriate.  An endpoint MUST NOT
+send further packets, which could result in a constant exchange of
+CONNECTION_CLOSE frames until the closing period on either peer ended.
 
```

Comment:
"An endpoint MUST NOT send further packets ...". Does that mean only the endpoint receiving the initial CLOSE frame, or both endpoints. In the latter case close retransmission is not possible, but that is inconsistent with other text.

Reply:
Each endpoint MUST NOT send further packets once they've received a CONNECTION_CLOSE, except to send a single CONNECTION_CLOSE of their own if desired.  Yes, that means that if the responding CONNECTION_CLOSE is dropped, it won't be retransmitted, but that's okay -- the other side is already closing, and won't be transmitting further packets to provoke a retransmission anyway.

Reply:
OK, I'll buy not sending more than one close after seeing one, although it could prolong the draining period in case of loss.

Reply:
I think actually it causes one of the endpoints to remain in closing instead of draining, but the spec says "The draining period SHOULD end when the closing period would have ended," so I think the time is the same whether you transition or not.